### PR TITLE
Improve processing of noisy benchmark results

### PR DESCRIPTION
**Pull request by janosg**
## Problem

Currently, the benchmark plots are based on the actual history of criterion values the optimizer generated. In the noisy case, this means that optimizers who get lucky draws are rewarded. 

## Solution

We re-evaluate a noise free criterion on the history of parameters the optimizer produced. The re-calculation is done in parallel to save time.

### CODE REVIEW

File: src/estimagic/benchmarking/run_benchmark.py [hunk 1]
```diff
@@ -154,40 +168,52 @@ def _get_kwargs_list_and_names(problems, opt_options):
...
+    Returns:
+        dict: Processed result.
+
+    """
+    _registry = get_registry(extended=True)
+    _criterion = problem["noise_free_criterion"]
+    _start_x = problem["inputs"]["params"]
+    _start_crit_value = _criterion(_start_x)
+    if isinstance(_start_crit_value, np.ndarray):
+        _start_crit_value = (_start_crit_value**2).sum()
```
Comment (timmens) [inline src/estimagic/benchmarking/run_benchmark.py:187]: If `_start_crit_value` is a `np.ndarray` we are always in the least-squares case?

File: src/estimagic/benchmarking/process_benchmark_results.py [hunk 2]
```diff
@@ -45,267 +42,146 @@ def create_convergence_histories(
...
-        converged = y_converged
-    elif stopping_criterion == "x":
-        converged = x_converged
-    elif stopping_criterion == "x_and_y":
-        converged = y_converged & x_converged
-    elif stopping_criterion == "x_or_y":
-        converged = y_converged | x_converged
+def _aggregate_idxs_with_or(x, y):
+    if x is None or y is None:
+        out = None
```
Comment (timmens) [inline src/estimagic/benchmarking/process_benchmark_results.py:184]: This should be `and` and then the else clause needs to be two elif clauses.

**Review by timmens — APPROVED** [review]
Conditional approve.

File: src/estimagic/benchmarking/process_benchmark_results.py (hunk 2 above)
Comment (janosg) [reply src/estimagic/benchmarking/process_benchmark_results.py]: Done.

**Pull request merged**